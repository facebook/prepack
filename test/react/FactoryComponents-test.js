/**
 * Copyright (c) 2017-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 */

/* @flow */

const setupReactTests = require("./setupReactTests");
const { runTest } = setupReactTests();

/* eslint-disable no-undef */
const { it } = global;

it("Simple factory classes", () => {
  runTest(__dirname + "/FactoryComponents/simple.js");
});

it("Simple factory classes 2", () => {
  runTest(__dirname + "/FactoryComponents/simple2.js");
});
