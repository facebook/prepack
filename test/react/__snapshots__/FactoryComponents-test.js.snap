// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple factory classes 2: (JSX => JSX) 1`] = `
ReactStatistics {
  "componentsEvaluated": 3,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "non-root factory class components are not suppoted",
          "name": "FactoryComponent",
          "status": "BAIL-OUT",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Simple factory classes 2: (JSX => createElement) 1`] = `
ReactStatistics {
  "componentsEvaluated": 3,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "non-root factory class components are not suppoted",
          "name": "FactoryComponent",
          "status": "BAIL-OUT",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Simple factory classes 2: (createElement => JSX) 1`] = `
ReactStatistics {
  "componentsEvaluated": 3,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "non-root factory class components are not suppoted",
          "name": "FactoryComponent",
          "status": "BAIL-OUT",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Simple factory classes 2: (createElement => createElement) 1`] = `
ReactStatistics {
  "componentsEvaluated": 3,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "message": "non-root factory class components are not suppoted",
          "name": "FactoryComponent",
          "status": "BAIL-OUT",
        },
      ],
      "message": "",
      "name": "App",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 2,
}
`;

exports[`Simple factory classes: (JSX => JSX) 1`] = `
ReactStatistics {
  "componentsEvaluated": 1,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [],
      "message": "",
      "name": "FactoryComponent",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;

exports[`Simple factory classes: (JSX => createElement) 1`] = `
ReactStatistics {
  "componentsEvaluated": 1,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [],
      "message": "",
      "name": "FactoryComponent",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;

exports[`Simple factory classes: (createElement => JSX) 1`] = `
ReactStatistics {
  "componentsEvaluated": 1,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [],
      "message": "",
      "name": "FactoryComponent",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;

exports[`Simple factory classes: (createElement => createElement) 1`] = `
ReactStatistics {
  "componentsEvaluated": 1,
  "evaluatedRootNodes": Array [
    Object {
      "children": Array [],
      "message": "",
      "name": "FactoryComponent",
      "status": "ROOT",
    },
  ],
  "inlinedComponents": 0,
  "optimizedNestedClosures": 0,
  "optimizedTrees": 1,
}
`;
